api/v1beta2/condition_types.go

```diff
@@ -19,15 +19,24 @@ package v1beta2
 const SourceFinalizer = "finalizers.fluxcd.io"
 
 const (
+	// ArtifactInStorageCondition indicates the availability of Artifact of the
+	// Source in the storage.
+	// If True, the Artifact is stored successfully. If False, the Artifact
+	// couldn't be stored.
+	// This Condition is only present on the resource if the Artifact is
+	// successfully stored.
```

Comment:
These two statements can come off as contradictory imo. The first statement says if artifact couldn't be stored, then a condition `ArtifactInStorage=False` would be present, but then the second statement  says `ArtifactInStorage` is only present if the artifact is stored.

Reply:
I think it's just meant to provide semantics of what it means and doesn't say that True or False will be present. The last statement describes when it'll be present. Not contradictory, but I'll remove it to be consistent with other conditions.